lib/exprengine.h

```diff
@@ -294,7 +294,7 @@ namespace ExprEngine {
         ValuePtr op1;
         ValuePtr op2;
     private:
-        std::string getName(const std::string &binop, ValuePtr op1, ValuePtr op2) const {
+        static std::string getName(const std::string& binop, ValuePtr op1, ValuePtr op2) {
```

Comment:
when this is non-static these parameters mask the equivalently named members. In general I think it's good to make things static where possible and also to avoid calling non-static methods during initializer lists where possible.